vllm/v1/worker/gpu_model_runner.py

```diff
@@ -495,7 +488,8 @@ def _update_states(self, scheduler_output: "SchedulerOutput") -> None:
                 mm_positions=new_req_data.mm_positions,
                 sampling_params=sampling_params,
                 pooling_params=pooling_params,
-                generator=generator,
+                # generator is populated in _update_sampling_states()
```

Comment:
I don't think its worth delaying this construction. This only happens once per request so I don't think the code readability change is worth the per win

Reply:
I guess it didn't really seem less readable to me - the generator is for seeded sampling so in some way fits naturally in `_update_sampling_states()`...

Reply:
I agree with @robertgshaw2-redhat
I think it's more natural to initialize it here.